addons/account/models/account_invoice.py

```diff
@@ -461,7 +461,7 @@ def compute_taxes(self):
     def unlink(self):
         for invoice in self:
             if invoice.state not in ('draft', 'cancel'):
-                raise UserError(_('You cannot delete an invoice which is not draft or cancelled. You should refund it instead.'))
+                raise UserError(_('You cannot delete an invoice which is not draft or cancelled. You should create Credit Note instead.'))
```

Comment:
You cannot delete an invoice which is not draft or cancelled. You should create a credit note instead.